tests/store/test_local_artifact_repo.py

```diff
@@ -61,6 +61,24 @@ def test_download_artifacts(local_artifact_repo):
         assert open(dst_path).read() == artifact_text
 
 
+def test_download_artifacts_does_not_copy(local_artifact_repo):
```

Comment:
Can we also add a test ensuring that `LocalArtifactRepository.download_artifacts()` *does* copy artifacts to `dst_path` when `dst_path` is not `None`?

Reply:
We already have coverage of that here right? https://github.com/mlflow/mlflow/pull/1307/files#diff-008ae91c4f707355b402124822ba93cdR96

Reply:
Not exactly. This test doesn't assert that `dst_path == dst_dir`.